app/api/routes/image_queries.py

```diff
@@ -186,13 +186,27 @@ async def post_image_query(
             if confidence_threshold is None:
                 confidence_threshold = detector_metadata.confidence_threshold  # Use detector's confidence threshold
 
+            mode = detector_metadata.mode
+            if mode == ModeEnum.BINARY:
+                result_type = ResultTypeEnum.binary_classification
+                results["label"] = "NO" if results["label"] else "YES"  # Map false / 0 to "YES" and true / 1 to "NO"
+            elif mode == ModeEnum.COUNT:
+                result_type = ResultTypeEnum.counting
+            elif mode == ModeEnum.MULTI_CLASS:
+                result_type = ResultTypeEnum.multi_classification
+            else:
+                raise ValueError(f"Got unrecognized detector mode: {mode}")
+
             image_query = create_iqe(
                 detector_id=detector_id,
+                result_type=result_type,
                 label=results["label"],
```

Comment:
This is going to change soon, you'll need to populate this field according to the mode type